src/protocol/mod.rs

```diff
@@ -194,12 +195,15 @@ impl Protocol {
     }
 
     fn add_arguments(&mut self, object: &Hash) -> R<()> {
-        if let Ok(arguments) = object.expect_field("arguments") {
-            self.arguments = arguments
-                .expect_str()?
-                .split_whitespace()
-                .map(String::from)
-                .collect();
+        if let Ok(arguments) = object
```

Comment:
here is where we call the new logic for parsing script arguments when creating a `Protocol`